playbooks/roles/csi-driver-config/tasks/install.yaml

```diff
@@ -0,0 +1,31 @@
+---
+# tasks file for csi driver installation
+
+- name: Setup csi driver dir
+  file:
+    path: "{{ csidir }}"
+    state: "{{ item }}"
+    mode: '0755'
+  with_items:
+  - directory
+
+- name: Downloading secret template file
+  get_url:
+    url: "{{ secret_template_url }}"
+    dest: "{{ csidir }}/secret.yaml"
+    mode: 0644
+
+- name: Update secret template file
+  shell: |
+    sed -i 's/""/"{{ ibmcloud_api_key }}"/' secret.yaml
```

Comment:
My concern is storing the api key on FS. @mkumatag

Reply:
this is not needed, may be you can update the content from the file and apply the spec on the fly